src/core/id.ts

```diff
@@ -9,38 +18,38 @@ import { CommandType, EventType } from './structures';
 export function reconstruct<T extends Interaction>(event: T) {
     switch (event.type) {
         case InteractionType.MessageComponent: {
-            return [`${event.customId}_C${event.componentType}`];
+            const data = parseParams(event, `_C${event.componentType}`)
+            return [data];
         }
         case InteractionType.ApplicationCommand:
-        case InteractionType.ApplicationCommandAutocomplete: {
-            return [`${event.commandName}_A${event.commandType}`, `${event.commandName}_B`];
-        }
+        case InteractionType.ApplicationCommandAutocomplete: 
+            return [{ id: `${event.commandName}_A${event.commandType}` }, { id: `${event.commandName}_B` }];
         //Modal interactions are classified as components for sern
         case InteractionType.ModalSubmit: {
-            return [`${event.customId}_M`];
+            const data = parseParams(event, '_M');
+            return [data];
         }
     }
 }
 /**
  *
  * A magic number to represent any commandtype that is an ApplicationCommand.
  */
-const appBitField = 0b000000001111;
+const PUBLISHABLE = 0b000000001111;
```

Comment:
are there any more constants like this? if yes then it'd be better to move it to a dedicated `constants.ts` file

Reply:
this should be the only one, afaik